lib/iiif_print/metadata.rb

```diff
@@ -0,0 +1,116 @@
+module IiifPrint
+  # rubocop:disable Metrics/ModuleLength, Metrics/ClassLength
+  class Metadata
+    def self.manifest_for(model:, version:, fields:)
+      new(model: model, version: version, fields: fields).manifest
+    end
+
+    def initialize(model:, version:, fields:)
+      @model = model
+      @version = version
+      @fields = fields
+    end
+
+    attr_reader :model, :version, :fields
+
+    # rubocop:disable MethodLength, CyclomaticComplexity, PerceivedComplexity, AbcSize
+    def manifest
+      case version
+      when 2
+        fields.map do |field|
+          options = field.options
+          parent = model.try(:model)
+          next if parent
+          label = Hyrax::Renderers::AttributeRenderer.new(field.name, nil).label
+          if field.name == :collection
+            next unless model[:member_of_collection_ids_ssim]&.present?
+            viewable_collections = Hyrax::CollectionMemberService.run(model, @current_ability)
+            next if viewable_collections.blank?
+            {
+              'label' => label,
+              'value' => make_collection_link(viewable_collections)
+            }
+          else
+            next if Array(model.try(field.name)).first.blank?
+            {
+              'label' => label,
+              'value' => cast_to_value(image: model, field_name: field.name, options: options)
+            }
+          end
+        end.compact
+      when 3
+        fields.map do |field|
+          values = Array(model.try(field.name)).map { |value| scrub(value.to_s) }
+          next if values.empty?
+          {
+            'label' => {
+              # Since we're using I18n to translate the field, we're setting the locale used in the translation.
+              I18n.locale.to_s => [Hyrax::Renderers::AttributeRenderer.new(field.name, nil).label]
+            },
+            'value' => {
+              "none" => values
+            }
+          }
+        end.compact
+      else
+        raise IiifPrintError, "Version #{version} not implemented"
+      end
+    end
+    # rubocop:enable MethodLength, CyclomaticComplexity, PerceivedComplexity, AbcSize
+
+    private
+
+    def scrub(value)
+      Loofah.fragment(value).scrub!(:whitewash).to_s
+    end
+
+    def cast_to_value(image:, field_name:, options:)
```

Comment:
I'm a bit confused we went from calling this a model to calling this an image.  I'd like to keep these consistent